### PR TITLE
Refactor store to chase wallet tip

### PR DESCRIPTION
**Pull request by quantumagi**
This PR:

- Refactors the store to chase the wallet tip instead of the chain tip.
- Update code to use IWithdrawalExtractor.
- Work around for DBreeze serialization issue when deserializing transactions.
- Refactors and extends test cases
- Remove duplication between test cases and merge into one file.

### CODE REVIEW

File: src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs [hunk 1]
```diff
@@ -667,27 +655,37 @@ private async Task OnInsertBlocksAsync(DBreeze.Transactions.Transaction dbreezeT
             // Find transfer transactions in blocks
             foreach (Block block in blocks)
             {
-                IReadOnlyList<IDeposit> deposits = this.depositExtractor.ExtractDepositsFromBlock(block, blockHeight);
+                uint256 blockHash = block.GetHash();
```
Comment (monsieurleberre) [inline src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs]: Then here I don't understand why you didn't use the existing ExtractWithdrawalsFromBlock(Block block, int blockHeight)?

File: src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/WithdrawalExtractor.cs [hunk 2]
```diff
@@ -54,7 +54,7 @@ public IReadOnlyList<IWithdrawal> ExtractWithdrawalsFromBlock(Block block, int b
             return withdrawalsFromBlock;
         }
```
Comment (monsieurleberre) [inline src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/WithdrawalExtractor.cs]: cf my comment above, I don't understand that change

File: src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs]: It is has an unwanted side-effect as discussed.

File: src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/WithdrawalExtractor.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/WithdrawalExtractor.cs]: see above reply.

**Pull request merged**